src/useReactions.tsx

```diff
@@ -285,6 +300,9 @@ export const ReactionsProvider = ({
       room.off(MatrixRoomEvent.Timeline, handleReactionEvent);
       room.off(MatrixRoomEvent.Redaction, handleReactionEvent);
       room.off(MatrixRoomEvent.LocalEchoUpdated, handleReactionEvent);
+      reactionTimeouts.forEach((t) => clearTimeout(t));
+      // If we're clearing timeouts, we also clear all reactions.
+      setReactions({});
     };
   }, [room, addRaisedHand, removeRaisedHand, memberships, raisedHands]);
```

Comment:
Since this effect depends on `memberships` and `raisedHands` it seems it will clear the reactions as soon as any of those change. The pattern I usually go for in this situation is wrapping the data in refs to make the effect less "reactive":

```typescript
const latestMemberships = useLatest(memberships)
useEffect(() => {
   // do something with latestMemberships.current
}, [latestMemberships])